bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/SharingStateService.kt

```diff
@@ -78,11 +75,11 @@ class SharingStateService(private val stateRepository: SharingStateRepository) {
         externalIds: Collection<String>?
     ): PageDto<SharingStateDto> {
 
-        val spec = Specification.allOf(byBusinessPartnerType(businessPartnerType), byExternalIdsIn(externalIds))
         val pageRequest = PageRequest.of(paginationRequest.page, paginationRequest.size)
-        val page = stateRepository.findAll(spec, pageRequest)
+        val spec = Specification.allOf(byExternalIdsIn(externalIds), byBusinessPartnerType(businessPartnerType))
+        val sharingStatePage = stateRepository.findAll(spec, pageRequest)
 
-        return page.toDto(page.content.map {
+        return sharingStatePage.toDto(sharingStatePage.content.map {
```

Comment:
Maybe use `.toPageDto { SharingStateDto(...) }` from `CommonMappings.kt`?

Reply:
I added it via a simple logic to map from business partner types to the relevant BPN.

Later on, we should move from setting BPN and business partner type explicitely in the sharing state and rather inferring it from the business partner stages. But this can only happen after we integrated the L/S/A endpoints to the generic model, made the external identifiers unique and the business partner type optional in the sharing state.